### PR TITLE
Use Failsafe to retry instead iceberg's Tasks in iceberg connector

### PR DESCRIPTION
**Pull request by krvikash**
## Description
Inspired from https://github.com/trinodb/trino/pull/14869#discussion_r1073208221

## Release notes

(X) This is not user-visible or docs only and no release notes are required.

### CODE REVIEW

**Review by alexjo2144 — COMMENTED** [review]
👍

**Comment by krvikash** [comment]
Rebased with upstream master.

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/AbstractIcebergTableOperations.java [hunk 1]
```diff
@@ -213,21 +215,20 @@ protected void refreshFromMetadataLocation(String newLocation)
...
-        Tasks.foreach(newLocation)
-                .retry(20)
-                .exponentialBackoff(100, 5000, 600000, 4.0)
-                .stopRetryOn(org.apache.iceberg.exceptions.NotFoundException.class) // qualified name, as this is NOT the io.trino.spi.connector.NotFoundException
-                .run(metadataLocation -> newMetadata.set(
-                        TableMetadataParser.read(fileIo, io().newInputFile(metadataLocation))));
+        TableMetadata newMetadata = Failsafe.with(new RetryPolicy<>()
+                        .withMaxRetries(20)
+                        .withBackoff(100, 5000, MILLIS, 4.0)
+                        .withMaxDuration(Duration.ofMillis(600000))
```
Comment (ebyhr) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/AbstractIcebergTableOperations.java:221]: Can we change to `Duration.ofMinutes` so that we can understand the duration quickly?

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/AbstractIcebergTableOperations.java [hunk 2]
```diff
@@ -213,21 +215,20 @@ protected void refreshFromMetadataLocation(String newLocation)
             return;
         }
 
-        AtomicReference<TableMetadata> newMetadata = new AtomicReference<>();
-        Tasks.foreach(newLocation)
-                .retry(20)
-                .exponentialBackoff(100, 5000, 600000, 4.0)
-                .stopRetryOn(org.apache.iceberg.exceptions.NotFoundException.class) // qualified name, as this is NOT the io.trino.spi.connector.NotFoundException
```
Comment (ebyhr) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/AbstractIcebergTableOperations.java:220]: Why did you remove the comment?

**Comment by krvikash** [comment]
Addressed comments.

**Comment by findepi** [comment]
Thanks!
